cloud/cluster/cluster.go

```diff
@@ -2,30 +2,31 @@ package cluster
 
 import (
 	"sort"
-	"time"
 )
 
 // Cluster represents a cluster of Nodes.
 type Cluster struct {
 	state    *state
 	reqCh    chan interface{}
-	ch       chan interface{}
-	updateCh chan []NodeUpdate
+	updateCh chan ClusterUpdate
 	subs     []chan []NodeUpdate
 }
 
+// Clusters can be updated in two ways:
+// *) a new state of the Cluster, which is a []Node
+// *) updates to specific Nodes, which is a []NodeUpdate
+type ClusterUpdate interface{}
```

Comment:
I think ClusterUpdate has to be private so that others can create it.

I'm happy to add func IsClusterUpdate(), but the downside is then you can't pass []Node, but Nodes([]Node{foo, bar})

If we think that's too ugly, I'm also alright with having one channel of []NodeUpdate and one of []Node, but having one of []Node and one of interface{} was weird.